phoenix-core/src/main/java/org/apache/phoenix/coprocessor/SystemCatalogRegionObserver.java

```diff
@@ -0,0 +1,59 @@
+/*
```

Comment:
We do not have a new coproc for updating the scan filter. We can add a boolean scan attribute for adding SystemCatalogViewIndexIdFilter and then update the scan filter in BaseScannerRegionObserver#preScannerOpen(). After updating the scan filter, you can set the attribute to the false to make sure that we update the scan filter once.

Reply:
After a side conversation with @yanxinyi, I realized that we need a new coproc in order to identify a syscat table region as we need to support old clients and we cannot expect on old client to set a new scan attributes.